include/ttmlir/Dialect/TTIR/IR/TTIROps.td

```diff
@@ -292,6 +292,42 @@ def TTIR_MatmulOp : TTIR_DPSOp<"matmul"> {
 }
 // ANCHOR_END: adding_an_op_matmul_ttir
 
+
+def TTIR_Conv2dOp : TTIR_DPSOp<"conv2d"> {
+    let summary = "Conv2d operation.";
+    let description = [{
+      Conv2d operation.
+    }];
+
+    let arguments = (ins AnyRankedTensor:$input,
+                         AnyRankedTensor:$weight,
+                         AnyRankedTensor:$bias,
+                         AnyRankedTensor:$output,
+                         SI32Attr:$kernel_height,
+                         SI32Attr:$kernel_width,
```

Comment:
We don't need to encode `kernel_height` or `kernel_width`, it can be inferred from the shape of the weights.

Reply:
I'm not sure I agree. Should we try to match pytorch api? As a model developer, it's much easier for me to just define kernel h/w and not think about the actual 4d shape of weights and the order of dims etc. (We'd have to change the weights to an optional tensor and do some validation in case user provides both weights and kernel dims).

Reply:
Oh right, but this isn't user facing, so scratch my comment...